src/program/render.rs

```diff
@@ -0,0 +1,225 @@
+use crate::infobar;
+use crate::program::{make_dst, Program};
+use sdl2::image::LoadTexture;
+use sdl2::pixels::Color;
+use sdl2::rect::Rect;
+
+const PADDING: i32 = 30;
+const HALF_PAD: i32 = 15;
+
+impl<'a> Program<'a> {
+    /// render loads the image at the path in the images path vector located at the index and
+    /// renders to screen
+    pub fn render(&mut self) -> Result<(), String> {
+        self.screen.canvas.set_draw_color(dark_grey());
+        if self.paths.images.is_empty() {
+            return self.render_blank();
+        }
+        self.screen.canvas.clear();
+        self.render_image()?;
+        if self.ui_state.render_infobar {
+            self.render_infobar()?;
+        }
+
+        // Present to screen
+        self.screen.canvas.present();
+        Ok(())
+    }
+
+    fn render_image(&mut self) -> Result<(), String> {
+        self.set_image_texture()?;
```

Comment:
Splitting the setting and rendering of the image into two steps will be useful later.

As an example, when adjusting the zoom, or toggling actual size of the image, there is no need to reload the texture (just used a cached version of the texture).

Is that what 
```r
        if self.paths.index == self.screen.last_index
            && self.screen.last_texture.is_some()
            && !self.screen.dirty
```

Is doing?

It won't reload the texture unless it needs to be reloaded, such as the image being deleted.

Reply:
Precisely. You can really see it in action when you have a large image, toggling the info bar was just as slow as loading a large image. Now, toggling the infobar on a large image is pretty much instantaneous.